samples/word/99-preview-apis/content-control-ondeleted-event.yaml

```diff
@@ -0,0 +1,161 @@
+id: word-content-control-ondeleted-event
+name: Content control onDeleted event
+description: Register and trigger onDeleted event on content controls.
+host: WORD
+api_set:
+    WordApi: '1.4'
+script:
+    content: |
+        $("#insert-content-controls").click(() => tryCatch(insertContentControls));
+        $("#register-event-handlers").click(() => tryCatch(registerEventHandlers));
+        $("#delete-content-control").click(() => tryCatch(deleteContentControl));
+        $("#setup").click(() => tryCatch(setup));
+
+        async function insertContentControls() {
+          // Traverses each paragraph of the document and wraps a content control on each.
+          await Word.run(async (context) => {
+            let paragraphs = context.document.body.paragraphs;
+            paragraphs.load("$none"); // Don't need any properties; just wrap each paragraph with a content control.
+
+            await context.sync();
+
+            for (let i = 0; i < paragraphs.items.length; i++) {
+              let contentControl = paragraphs.items[i].insertContentControl();
+              contentControl.tag = "forTesting";
+            }
+
+            console.log("Content controls inserted: " + paragraphs.items.length);
+
+            await context.sync();
+          });
+        }
+
+        async function registerEventHandlers() {
+          await Word.run(async (context) => {
+            let contentControls = context.document.contentControls;
+            contentControls.load("items");
+            await context.sync();
+
+            // Register the onDeleted event handler on each content control.
+            if (contentControls.items.length === 0) {
+              console.log("There aren't any content controls in this document so can't register event handlers.");
+            } else {
+              for (let i = 0; i < contentControls.items.length; i++) {
+                contentControls.items[i].onDeleted.add(contentControlDeleted);
+                contentControls.items[i].track();
```

Comment:
Why are these being manually tracked?

Reply:
This was recommended by the engineer. In this sample, the event doesn't fire (or least the handler isn't run) without that statement.

Reply:
Hmm, I didn't encounter that issue (works fine for me without the tracking). If it is necessary, I'd recommend adding a comment explaining why it's needed.

Reply:
Nevermind, I'm mistaken. I do think we should add a note explaining why that `track` statement is needed, since it deviates from the Excel event pattern.